src/Features/CSharp/Portable/Completion/CompletionProviders/OperatorIndexerCompletionProvider.cs

```diff
@@ -0,0 +1,234 @@
+                };
+                if (completionChange is { })
+                {
+                    return completionChange;
+                }
+            }
+
+            return await base.GetChangeAsync(document, item, completionListSpan, commitKey, disallowAddingImports, cancellationToken).ConfigureAwait(false);
+        }
+
+        private static async Task<CompletionChange?> HandleConversionChangeAsync(Document document, CompletionItem item, CancellationToken cancellationToken)
+        {
+            var position = SymbolCompletionItem.GetContextPosition(item);
+            if (!item.Properties.TryGetValue(MinimalTypeNamePropertyName, out var typeName))
+            {
+                return null;
+            }
+
+            var root = await document.GetRequiredSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+            var tokenAtPosition = root.FindTokenOnLeftOfPosition(position);
+            var normalizedToken = tokenAtPosition.GetPreviousTokenIfTouchingWord(position);
+            // syntax tree manipulations are to complicated if a mixture of conditionals is involved. Some text manipulation is easier here.
+            //                      ↓               | cursor position
+            //                   ↓                  | normalizedToken (dot)
+            // white?.Black.White.Black?.White      | current user input
+            // white?.Black.White.Black?.White      | rootExpression (text manipulation starts with this)
+            //       .Black.White                   | parentExpression (needed to calculate the position to insert the closing brace)
+            //                    Black             | identifier at cursor position (gets removed, because the user typed the name of a type)
+            // |----------------------|             | part to replace (TextChange.Span), if identifier is not present: ends at rootExpression.End (after White.)
+            //                   ↑                  | insert closing brace between White and dot (parentExpression.Span.End)
+            // ((Black)white?.Black.White).?.White  | The result. Because we removed the identifier, the remainder after the identifier may be syntactically wrong 
+            //                             ↑        | cursor after the manipulation is placed after the dot
+            var rootExpression = GetRootExpressionOfInvocation(normalizedToken);
+            var parentExpression = GetParentExpressionOfInvocation(normalizedToken);
+            var identifier = tokenAtPosition.Parent as IdentifierNameSyntax;
+            if (rootExpression is null || parentExpression is null)
+            {
+                return null;
+            }
+
+            var spanToReplace = TextSpan.FromBounds(rootExpression.Span.Start, identifier is null ? rootExpression.Span.End : identifier.Span.End);
+            var cursorPositionOffset = spanToReplace.End - position;
+            var fromRootToParent = rootExpression.ToString();
```

Comment:
this will cut off preceding trivia of hte root expression. is that what you want?

Reply:
Trivia is kept. I have some trivia tests in place:
https://github.com/dotnet/roslyn/pull/47511/files#diff-5503a64e2b97282b2362dc32e1b9423bR639-R646